### PR TITLE
Switch from React Hot Loader to React Refresh

### PR DESCRIPTION
**Pull request by jcq**
### Description
React-Hot-Loader [is being deprecated](https://github.com/gaearon/react-hot-loader#moving-towards-next-step) favor of React Fast Refresh. It is recommended to move to using the latter if environment supports it (ours does).

This also sets headers for webpack-dev-server so HMR works correctly when running rspec feature tests.

### Acceptance Criteria
- [ ] Remove `react-hot-loader`
- [ ] Add React Fast Refresh and configure Webpack appropriately
- [ ] Code compiles correctly
- [ ] Hot reloading still works as intended when in dev mode
- [ ] Hot reloading now properly works during rspec feature tests

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "build:demo": "NODE_ENV=production yarn run build && yarn run build:storybook",
     "dev:clear-build": "rm ../app/assets/javascripts/*webpack* || true",
     "dev": "yarn run dev:clear-build && NODE_ENV=development yarn run build -w",
-    "dev:hot": "yarn run dev:clear-build && NODE_ENV=development node webpack-hot-server.js",
+    "dev:hot": "webpack-dev-server",
```
Comment (meckila-navapbc) [inline client/package.json:26]: ➕1️⃣ For the script tweak!

File: client/webpack.config.js [hunk 2]
```diff
@@ -1,20 +1,41 @@
...
   output: {
     filename: 'webpack-bundle.js',
     sourceMapFilename: 'sourcemap-[file].map',
-    path: path.join(__dirname, '../app/assets/javascripts')
+    path: path.join(__dirname, '../app/assets/javascripts'),
+    publicPath: 'http://localhost:3500/',
+  },
+  plugins: [
+    new webpack.EnvironmentPlugin({ NODE_ENV: 'development' }),
+    // devBuild && new webpack.HotModuleReplacementPlugin(),
```
Comment (meckila-navapbc) [inline client/webpack.config.js:19]: Is this needed still?

File: client/webpack.config.js [hunk 3]
```diff
@@ -1,20 +1,41 @@
...
+    devBuild && !testBuild && new ReactRefreshWebpackPlugin(),
+  ].filter(Boolean),
+  devServer: {
+    // hmr: true,
+    headers: {
+      'Access-Control-Allow-Origin': '*',
+      'Access-Control-Allow-Credentials': 'true',
+    },
+    hotOnly: true,
```
Comment (meckila-navapbc) [inline client/webpack.config.js:28]: [Is this a solution for hot module replacement not getting detected](https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/main/docs/TROUBLESHOOTING.md#hot-module-replacement-hmr-is-not-enabled)?

And if so, can the commented out `hmr` be removed?

Comment (jcq) [reply client/webpack.config.js:26]: `hotOnly` indicates that it should do HMR only and not fall back to refreshing the entire page

But yeah, the `hmr` key is not needed, and I've removed

File: client/webpack.config.js (hunk 2 above)
Comment (jcq) [reply client/webpack.config.js]: Nope! Removing the commented line.

**Review by meckila-navapbc — APPROVED** [review]

**Pull request merged**